src/modules/ethereum/message-signing/prepareTypedData.ts

```diff
@@ -12,7 +12,12 @@ export function isPermit({ message }: TypedData) {
 export function toTypedData(data: string | Partial<TypedData>): TypedData {
   if (typeof data === 'string') {
     try {
-      const typedData = JSON.parse(data);
+      const typedData = JSON.parse(data) as TypedData;
+      // Backend expect chainId to be a string
+      // TODO: remove when backend fixed
+      typedData.domain.chainId = typedData.domain.chainId
+        ? String(typedData.domain.chainId)
+        : undefined;
```

Comment:
1. remove?
2. this is safer:
```suggestion
      // TODO: remove when backend fixed
      if (typedData.domain.chainId) {
        typedData.domain.chainId = String(typedData.domain.chainId)
      }
```